File: src/main/java/com/mparticle/kits/MParticlePushProvider.java
package com.mparticle.kits;

import android.content.Context;
import androidx.annotation.NonNull;

import com.urbanairship.UAirship;
import com.urbanairship.push.PushMessage;
import com.urbanairship.push.PushProvider;

/**
 * Used to register for push in the Urban Airship SDK.
 */
class MParticlePushProvider implements PushProvider {

    private static MParticlePushProvider instance = new MParticlePushProvider();
    private String token;

    public static MParticlePushProvider getInstance() {
        return instance;
    }

    private MParticlePushProvider() {}

    @Override
    public int getPlatform() {
        return UAirship.ANDROID_PLATFORM;
    }

    @androidx.annotation.NonNull
    @Override
    public String getDeliveryType() {
        return PushProvider.FCM_DELIVERY_TYPE;
    }

    @Override
    public String getRegistrationToken(@NonNull Context context) {
        return token;
    }

    @Override
    public boolean isAvailable(@NonNull Context context) {
        return true;
    }

    @Override
    public boolean isSupported(@NonNull Context context) {
        return true;
    }

    void setRegistrationToken(String token) {
        this.token = token;
    }
}

File: src/main/kotlin/com/mparticle/kits/MParticlePushProvider.kt
package com.mparticle.kits

import android.content.Context
import com.urbanairship.UAirship
import com.urbanairship.push.PushProvider

/**
 * Used to register for push in the Urban Airship SDK.
 */
internal class MParticlePushProvider private constructor() : PushProvider {
    private var token: String? = null

    override fun getPlatform(): Int = UAirship.ANDROID_PLATFORM

    override fun getDeliveryType(): String = PushProvider.FCM_DELIVERY_TYPE

    override fun getRegistrationToken(context: Context): String? = token

    override fun isAvailable(context: Context): Boolean = true

    override fun isSupported(context: Context): Boolean = true

    fun setRegistrationToken(token: String?) {
        this.token = token
    }

    companion object {
        val instance = MParticlePushProvider()
    }
}
